src/lib.rs

```diff
@@ -225,18 +227,22 @@ cfg_if! {
                   target_env = "sgx",
               )))] {
         #[path = "rdrand.rs"] mod imp;
-    } else if #[cfg(target_arch = "wasm32")] {
-        cfg_if! {
-            if #[cfg(feature = "wasm-bindgen")] {
-                #[path = "wasm32_bindgen.rs"] mod imp;
-            } else if #[cfg(feature = "stdweb")] {
-                #[path = "wasm32_stdweb.rs"] mod imp;
-            } else {
-                #[path = "dummy.rs"] mod imp;
-            }
-        }
-    } else {
+    // the following two branches are intended only for `wasm32-unknown-unknown`
+    // target and may not work or work inefficiently on targets which may be
+    // added in future
+    } else if #[cfg(all(target_arch = "wasm32", feature = "wasm-bindgen"))] {
+        #[path = "wasm32_bindgen.rs"] mod imp;
+    } else if #[cfg(all(target_arch = "wasm32", feature = "stdweb"))] {
+        #[path = "wasm32_stdweb.rs"] mod imp;
+    } else if #[cfg(feature = "dummy")] {
         #[path = "dummy.rs"] mod imp;
+    } else {
+        compile_error!("\
+            target is not supported, you may enable dummy implementation \
+            using the `dummy` feature or overwrite `getrandom` crate with \
+            a custom one which supports your target using `[replace]` or \
```

Comment:
Can we make this shorter? E.g.

> target not supported; you may enable compile-only support via `dummy` feature or see getrandom#4 for custom target support

Reply:
Maybe we should add a section to docs regarding handling of unsupported targets and link it in this message? So it will look like this:
> target is not supported, for more information see: https://docs.rs/getrandom/#unsupported-targets